gui/gui/src/ts/components/search/NodeSearchBox.tsx

```diff
@@ -121,7 +124,9 @@ export const NodeSearchBox: React.FunctionComponent<{
     // Call throttled query with new search text
     throttledQuery.current(
       {
-        text: search.text,
+        text: `label:${search.text} ${datasources
```

Comment:
I'm not sure about this.
1) We want to allow the search box text input to accept Lucene queries, which can be pretty complex.
2) There is an additional attribute, searching by node id (id:), which would not work with this code.

The thing to do in this situation is to have two inputs: fulltext search query as text and a separate structured "filters" object that we pass to the backend. Then getMatchingNodes(text, filters: {datasources: [...]}). 

(There will be another options: input to do prefix instead of fulltext search.)

Just to make it general out of the gate, because we'll need it, the filters should be filters: {datasources: include: [...], exclude: [...]}).

I've done this in other systems. See here
https://github.com/minorg/paradicms/tree/master/lib/scala/generic/src/main/scala/org/paradicms/lib/generic/stores
for the paradicms implementation of faceted search.
As with Lucene or ElasticSearch, the filters override / are applied after the fulltext search.

Complex inputs in GraphQL are kind of a pain, because Sangria doesn't nicely deserialize objects. Here's the ugly version of input object types in paradicms:
https://github.com/minorg/paradicms/blob/master/app/generic/app/models/graphql/GenericGraphQlSchemaDefinition.scala

I found some magic I use in another project but haven't done in paradicms yet:
https://github.com/minorg/CollaborationNetworks/blob/master/app/app/models/graphql/GraphQlSchemaDefinition.scala
(You'll need the circe dependencies from the build.sbt.)

@123joshuawu If you want to tackle that go ahead. Can also wait for another PR.

Reply:
I filed a separate issue #150, leave this implementation for now.